wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUXRefClient.java

```diff
@@ -0,0 +1,557 @@
+     * @see org.hpccsystems.ws.client.BaseHPCCWsClient#getDefaultStub()
+     */
+    @Override
+    public Stub getDefaultStub() throws AxisFault
+    {
+        return new WsDFUXRefStub();
+    }
+
+    /**
+     * Gets the.
+     *
+     * @param connection
+     *            the connection
+     * @return the HPCC WSDFUXREF client
+     */
+    public static HPCCWsDFUXRefClient get(Connection connection)
+    {
+        return new HPCCWsDFUXRefClient(connection);
+    }
+
+    /**
+     * Gets the.
+     *
+     * @param protocol
+     *            the protocol
+     * @param targetHost
+     *            the target host
+     * @param targetPort
+     *            the target port
+     * @param user
+     *            the user
+     * @param pass
+     *            the pass
+     * @return the HPCC WSDFUXREF client
+     */
+    public static HPCCWsDFUXRefClient get(String protocol, String targetHost, String targetPort, String user, String pass)
+    {
+        Connection conn = new Connection(protocol, targetHost, targetPort);
+        conn.setCredentials(user, pass);
+        return new HPCCWsDFUXRefClient(conn);
+    }
+
+    /**
+     * Gets the.
+     *
+     * @param protocol
+     *            the protocol
+     * @param targetHost
+     *            the target host
+     * @param targetPort
+     *            the target port
+     * @param user
+     *            the user
+     * @param pass
+     *            the pass
+     * @param timeout
+     *            the timeout
+     * @return the HPCC WSDFUXREF client
+     */
+    public static HPCCWsDFUXRefClient get(String protocol, String targetHost, String targetPort, String user, String pass, int timeout)
+    {
+        Connection conn = new Connection(protocol, targetHost, targetPort);
+        conn.setCredentials(user, pass);
+        conn.setConnectTimeoutMilli(timeout);
+        conn.setSocketTimeoutMilli(timeout);
+
+        return new HPCCWsDFUXRefClient(conn);
+    }
+
+    /**
+     * Instantiates a new HPCC WSDFUXREF client.
+     *
+     * @param baseConnection
+     *            the base connection
+     */
+    protected HPCCWsDFUXRefClient(Connection baseConnection)
+    {
+        initWsDFUXRefClientStub(baseConnection);
+    }
+
+    /**
+     * Initializes the service's underlying stub Should only be used by constructors.
+     *
+     * @param conn
+     *            -- All connection settings included
+     */
+    protected void initWsDFUXRefClientStub(Connection conn)
+    {
+        try
+        {
+            setActiveConnectionInfo(conn);
+            stub = setStubOptions(new WsDFUXRefStub(conn.getBaseUrl() + WSDFUXREFURI), conn);
+        }
+        catch (AxisFault e)
+        {
+            log.error("Could not initialize WsDFUXRefStub - Review all HPCC connection values");
+            e.printStackTrace();
+        }
+        catch (Exception e)
+        {
+            log.error("Could not initialize WsDFUXRefStub - Review all HPCC connection values");
```

Comment:
Same question as above. Do we need to invalidate the client somehow here?

Reply:
ensuring the stub is null, and populating the initerr message will invalidate client.